backend/src/routes/penRequest.js

```diff
@@ -48,48 +42,8 @@ router.post('/reject', passport.authenticate('jwt', {session: false}, undefined)
 router.post('/return', passport.authenticate('jwt', {session: false}, undefined), auth.isValidAdminToken, returnPenRequest);
 
 // retrieve all the documents.
-router.get('/:id/documents', passport.authenticate('jwt', {session: false}, undefined), auth.isValidAdminToken,
-  async (req, res) => {
-    const token = utils.getBackendToken(req);
-    axios.defaults.headers['common']['Authorization'] = `Bearer ${token}`;
-    const url = `${config.get('server:penRequestURL')}/${req.params.id}/documents`;
-    log.debug(`url is ${url}`);
-    axios.get(url).then(documentResponse =>{
-      const results=[];
-      for(const element of documentResponse.data)  {
-        if(element.fileSize)
-          element.fileSize = file.humanFileSize(element.fileSize);
-        results.push(element);
-      }
-      return res.status(200).json(results);
-    }).catch(error=>{
-      log.error('An error occurred attempting to get documents.');
-      log.error(error);
-      log.error(JSON.stringify(error.response.data));
-      return res.status(500).json();
-    });
-  });
+router.get('/:id/documents', passport.authenticate('jwt', {session: false}, undefined), auth.isValidAdminToken, getDocuments);
+
 // retrieve the document by its id.
-router.get('/:id/documents/:documentId', auth.isValidAdminToken,
-  async (req, res) => {
-    const token = utils.getBackendToken(req);
-    if(!token){
-      return res.status(401).json({
-        status: 401,
-        message: 'You are not authorised to access this page.'
-      });
-    }
-    axios.defaults.headers['common']['Authorization'] = `Bearer ${token}`;
-    const url = `${config.get('server:penRequestURL')}/${req.params.id}/documents/${req.params.documentId}`;
-    log.debug(`url is ${url}`);
-    axios.get(url).then(resData =>{
-      res.setHeader('Content-disposition', 'attachment; filename=' + resData.data.fileName);
-      res.setHeader('Content-type', resData.data.fileExtension);
-      return res.status(200).send(Buffer.from(resData.data.documentData, 'base64'));
-    }).catch(error=>{
-      log.error('An error occurred attempting to get documents.');
-      log.error(error);
-      return res.status(500).json();
-    });
-  });
+router.get('/:id/documents/:documentId', auth.isValidAdminToken, getDocumentById);
```

Comment:
should authenticate jwt

Reply:
Because browsers can directly download files through the url, we don't need to use axios to get file data on the client side. The download url is still protected by the session through auth.isValidAdminToken, only user logged in can download files.